cmake/Options/TileDBToolchain.cmake

```diff
@@ -33,11 +58,20 @@ if (NOT DEFINED CMAKE_TOOLCHAIN_FILE)
     endif()
 endif()
 
-if(APPLE AND NOT DEFINED VCPKG_TARGET_TRIPLET)
-    if (CMAKE_OSX_ARCHITECTURES STREQUAL x86_64 OR CMAKE_SYSTEM_PROCESSOR MATCHES "(x86_64)|(AMD64|amd64)|(^i.86$)")
-        set(VCPKG_TARGET_TRIPLET "x64-osx")
-    elseif (CMAKE_OSX_ARCHITECTURES STREQUAL arm64 OR CMAKE_SYSTEM_PROCESSOR MATCHES "^aarch64" OR CMAKE_SYSTEM_PROCESSOR MATCHES "^arm")
-        set(VCPKG_TARGET_TRIPLET "arm64-osx")
+if(NOT DEFINED VCPKG_TARGET_TRIPLET AND SANITIZER STREQUAL "address")
```

Comment:
> sanitizer-relevant code in two places rather than one

The `Sanitizer` module has the code for setting the compiler options when building our own code, and `TileDBToolChain` has the code for choosing the vcpkg triplet to use. I would prefer these two to stay in separate places, especially as the triplet selection might become more complex in the future.